api/app/GraphQL/Mutations/CreateOrUpdateScreeningQuestionAssessmentStep.php

```diff
@@ -58,11 +58,12 @@ public function __invoke($_, array $args)
 
                     $questionToUpdate->save();
                 } else {
-                    $pool->screeningQuestions()->create([
-                        'assessment_step_id' => $assessmentStep->id,
-                        'question' => $incomingQuestion['question'],
-                        'sort_order' => isset($incomingQuestion['sortOrder']) ? $incomingQuestion['sortOrder'] : null,
-                    ]);
+                    $newQuestion = new ScreeningQuestion;
+                    $newQuestion->assessment_step_id = $assessmentStep->id;
+                    $newQuestion->question = $incomingQuestion['question'];
+                    $newQuestion->sort_order = isset($incomingQuestion['sortOrder']) ? $incomingQuestion['sortOrder'] : null;
+                    $newQuestion->save();
+                    $pool->screeningQuestions()->save($newQuestion);
```

Comment:
Here's how you can make this work without making pool_id nullable.
```suggestion
                    $newQuestion = new ScreeningQuestion();
                    $newQuestion->assessment_step_id = $assessmentStep->id;
                    $newQuestion->question = $incomingQuestion['question'];
                    $newQuestion->sort_order = isset($incomingQuestion['sortOrder']) ? $incomingQuestion['sortOrder'] : null;
                    $pool->screeningQuestions()->save($newQuestion);
```

Reply:
Oh cool, thanks I didn't know you could do that. 
I implemented both things in https://github.com/GCTC-NTGC/gc-digital-talent/pull/9161/commits/727e790fdd913738b717466744d3f48c181c88b1